### PR TITLE
[Docs] .server modules and .client modules

### PR DESCRIPTION
**Pull request by pierophp**
I had a [conversation ](https://bsky.app/profile/pierophp.bsky.social/post/3lco2fpgbhk2h)with @brookslybrand on Bluesky, and I'm adding the Remix .server modules [documentation](https://remix.run/docs/en/main/file-conventions/-server#server-modules) to [React Router](https://reactrouter.com/explanation/special-files).

Thank you

### CODE REVIEW

**Comment by brookslybrand** [comment]
Thanks @pierophp!

Will you also add [.client modules](https://remix.run/docs/en/main/file-conventions/-client) as well for completion sake.

Also, be sure to move these sections above all of the links defined at the bottom

**Comment by pierophp** [comment]
Done @brookslybrand

File: docs/explanation/special-files.md [hunk 1]
```diff
@@ -281,6 +281,48 @@ For an example, please refer to the default [`entry.server.tsx`][node-streaming-
...
+
+```txt
+app
+├── .server 👈 marks all files in this directory as server-only
+│   ├── auth.ts
+│   └── db.ts
+├── cms.server.ts 👈 marks this file as server-only
+├── root.tsx
+└── routes
+    └── _index.tsx
```
Comment (brookslybrand) [inline docs/explanation/special-files.md:297]:
```suggestion
└── routes.ts
```

File: docs/explanation/special-files.md [hunk 2]
```diff
@@ -281,6 +281,48 @@ For an example, please refer to the default [`entry.server.tsx`][node-streaming-
...
+
+```ts
+import { supportsVibrationAPI } from "./feature-check.client.ts";
+
+console.log(supportsVibrationAPI);
+// server: undefined
+// client: true | false
+```
+
+Refer to the Route Module section in the sidebar for more information.
```
Comment (brookslybrand) [inline docs/explanation/special-files.md:324]:
```suggestion
```

**Review by brookslybrand — APPROVED** [review]
Thanks @pierophp for adding these! I just had some very minor things I wanted changed, so I went ahead and adjusted those

**Pull request merged**